corgidrp/data.py

```diff
@@ -184,6 +185,57 @@ def rescale_error(self, input_error, err_name):
         for i, frame in enumerate(self.frames):
             frame.err = self.all_err[i]               
 
+    def split_dataset(self, prihdr_keywords=None, exthdr_keywords=None):
+        """
+        Splits up this dataset into multiple smaller datasets that have the same set of header keywords
+        The code uses all keywords together to determine an unique group
+
+        Args: 
+            prihdr_keywords (list of str): list of primary header keywords to split
+            exthdr_keywords (list of str): list of 1st extension header keywords to split on
+
+        Returns:
+            list of datasets: list of sub datasets
+            list of tuples: list of each set of unique header keywords. pri_hdr keywords occur before ext_hdr keywords
+        """
+        if prihdr_keywords is None and exthdr_keywords is None:
+            raise ValueError("No prihdr or exthdr keywords passed in to split dataset")
+        
+        col_names = []
+        col_vals = []
+        if prihdr_keywords is not None:
+            for key in prihdr_keywords:
+                dataset_vals = [frame.pri_hdr[key] for frame in self.frames]
```

Comment:
what if the key is present in some but not all files? is that too much of an edge case to care about? we might want to be error tolerant to a key not existing though.

Reply:
It'll throw an error here that the key doesn't exist. I think that's what we want since that means we can't split the dataset on this entry because some of the data is missing it. Likely points to a problem with the data.